packages/rocketchat-lib/server/lib/sendEmailOnMessage.js

```diff
@@ -114,14 +114,33 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
 					return;
 				}
 
+				// Footer in case direct reply is enabled.
+				if (RocketChat.settings.get('Direct_Reply_Enable')) {
+					footer = RocketChat.placeholders.replace(RocketChat.settings.get('Email_Footer_Direct_Reply') || '');
+				}
+
 				user.emails.some((email) => {
 					if (email.verified) {
-						email = {
-							to: email.address,
-							from: RocketChat.settings.get('From_Email'),
-							subject: `[${ siteName }] ${ emailSubject }`,
-							html: header + messageHTML + divisorMessage + (linkByUser[user._id] || defaultLink) + footer
-						};
+						// If direct reply enabled, email content with headers
+						if (RocketChat.settings.get('Direct_Reply_Enable')) {
+							email = {
+								to: email.address,
+								from: RocketChat.settings.get('From_Email'),
+								subject: `[${ siteName }] ${ emailSubject }`,
+								headers: {
+									// Reply-To header with format "username+messageId@domain"
+									'Reply-To': `${ RocketChat.settings.get('Direct_Reply_Username').split('@')[0] }+${ message._id }@${ RocketChat.settings.get('Direct_Reply_Username').split('@')[1] }`
+								},
+								html: header + messageHTML + divisorMessage + (linkByUser[user._id] || defaultLink) + footer
+							};
+						} else {
+							email = {
+								to: email.address,
```

Comment:
You can declare this before the if and just add the header property inside the if